db/migrate/20210326172406_add_published_index_to_listings.rb

```diff
@@ -0,0 +1,15 @@
+class AddPublishedIndexToListings < ActiveRecord::Migration[6.0]
+  disable_ddl_transaction!
+
+  def up
+    unless index_exists?(:classified_listings, :published)
+      add_index :classified_listings, :published, algorithm: :concurrently
+    end
+  end
+
+  def down
+    if index_exists?(:classified_listings, :published)
+      remove_index :classified_listings, column: :published, algorithm: :concurrently
+    end
+  end
```

Comment:
We've had deployment issues where the process would fail at some point and a restart caused the migration to run a second time, throwing an error because without the check it says "already exists", therefore blocking deployments. Several migrations have this check included. Rails 6.1 introduces more "native syntax" for these checks which will be nice.

Reply:
Ah, makes sense. One of the downsides of having to create indexes outside of a transaction. 😕

In the past I've only seen this in migrations that create multiple indexes and one of them added a `UNIQUE` constraint but with the problems with Heroku deploys earlier this week (their backup process blocked _all_ DDL queries), I can totally see this being more of a problem even on singular indexes with no constraints.